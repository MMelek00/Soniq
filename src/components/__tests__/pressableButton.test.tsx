import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import PressableButton from '../pressableButton';
describe('PressableButton', () => {
  const mockPress = jest.fn();
  const title = 'Press me!';

  const renderButton = () => {
    return render(
      <PressableButton
        bgColor={'darkslateblue'}
        onPress={mockPress}
        title={title}
      />,
    );
  };
  it('renders correctly', () => {
    const tree = renderer.create(
      <PressableButton
        bgColor={'darkslateblue'}
        onPress={mockPress}
        title="Do something"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should find the button title', () => {
    const {getByText} = renderButton();

    const foundButtonTitle = getByText(title);

    expect(foundButtonTitle.props.children).toEqual(title);
  });
  it('onPress work correctly', () => {
    const {getByTestId} = renderButton();

    const MockButton = getByTestId('PressableButton');
    fireEvent.press(MockButton);
    expect(mockPress).toHaveBeenCalled();
  });
});
