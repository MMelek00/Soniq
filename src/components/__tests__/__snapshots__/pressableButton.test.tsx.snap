// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "darkslateblue",
      "borderRadius": 10,
      "marginVertical": 20,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
    }
  }
  testID="PressableButton"
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "textAlign": "center",
        },
      ]
    }
  >
    Do something
  </Text>
</View>
`;
